sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/SubscribeTransform.java

```diff
@@ -134,19 +160,28 @@ private TopicPath getTopicPath() {
     }
   }
 
-  @Override
-  public PCollection<SequencedMessage> expand(PBegin input) {
-    PCollection<SubscriptionPartition> subscriptionPartitions;
-    subscriptionPartitions =
+  @SuppressWarnings("unused")
+  private PCollection<SequencedMessage> expandSdf(PBegin input) {
+    PCollection<SubscriptionPartition> subscriptionPartitions =
         input.apply(new SubscriptionPartitionLoader(getTopicPath(), options.subscriptionPath()));
-
     return subscriptionPartitions.apply(
         ParDo.of(
             new PerSubscriptionPartitionSdf(
-                new ManagedBacklogReaderFactoryImpl(this::newBacklogReader),
+                new ManagedFactoryImpl<>(this::newBacklogReader),
+                new ManagedFactoryImpl<>(this::newCommitter),
                 this::newInitialOffsetReader,
                 this::newRestrictionTracker,
-                this::newPartitionProcessor,
-                this::newCommitter)));
+                this::newPartitionProcessor)));
+  }
+
+  private PCollection<SequencedMessage> expandSource(PBegin input) {
+    return input.apply(
+        Read.from(
+            new UnboundedSourceImpl(options, this::newBufferedSubscriber, this::newBacklogReader)));
+  }
+
+  @Override
+  public PCollection<SequencedMessage> expand(PBegin input) {
+    return expandSource(input);
```

Comment:
option to use SDF or Source? Or can SDF be used elsewhere by configuration?

Reply:
This will be deferred to a child CL which will override on runnerv2